src/middlewared/middlewared/service.py

```diff
@@ -871,6 +873,74 @@ def ping(self):
         """
         return 'pong'
 
+    def _ping_host(self, host, timeout, counter=3):
+        if osc.IS_LINUX:
+            process = run(['ping', '-4', '-w', f'{timeout}', '-c', f'{counter}', host])
+        else:
+            process = run(['ping', '-t', f'{timeout}', '-c', f'{counter}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    def _ping6_host(self, host, timeout, counter=3):
+        if osc.IS_LINUX:
+            process = run(['ping6', '-w', f'{timeout}', '-c', f'{counter}', host])
+        else:
+            process = run(['ping6', '-X', f'{timeout}', '-c', f'{counter}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    @accepts(
+        Dict(
+            'options',
+            Str('type', enum=['ICMP', 'ICMPV4', 'ICMPV6'], default='ICMP'),
+            Str('hostname', required=True),
+            Int('timeout', validators=[Range(min=1, max=60)], default=10),
+        ),
+    )
+    def ping_remote(self, options):
+        """
+        Method that will send an ICMP echo request to "hostname"
+        and will wait up to "timeout" for a reply.
+        """
+        ip = None
+        ip_found = True
+        try:
+            ip = IpAddress()
+            ip(options['hostname'])
+            ip = options['hostname']
+        except ValueError:
+            ip_found = False
+        if not ip_found:
+            try:
+                if options['type'] == 'ICMP':
+                    ip = socket.getaddrinfo(options['hostname'], None)[0][4][0]
+                elif options['type'] == 'ICMPV4':
+                    ip = socket.getaddrinfo(options['hostname'], None, socket.AF_INET)[0][4][0]
+                elif options['type'] == 'ICMPV6':
+                    ip = socket.getaddrinfo(options['hostname'], None, socket.AF_INET6)[0][4][0]
+                else:
+                    return False
+            except socket.gaierror:
+                return False
+
+        addr = ipaddress.ip_address(ip)
+        ping_host = False
+        if addr.version == 4 and (options['type'] == 'ICMPV4' or options['type'] == 'ICMP'):
```

Comment:
Sorry, something more came up in last review after your changes.

If you specify ICMPV4 and give a IPV6 it should error out, give a validation error. Not simply return False as it does right now.

Once you add the validation the second part of these both if statements wont be necessary.